tests/apps/configurationapp/app.go

```diff
@@ -0,0 +1,274 @@
+	}
+	return subscriptionId, nil
+}
+
+// startSubscription is the handler for starting a subscription to config store
+func startSubscription(w http.ResponseWriter, r *http.Request) {
+	var keys []string
+	err := json.NewDecoder(r.Body).Decode(&keys)
+	if err != nil {
+		sendResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	subscriptionId, err := subscribeToConfigUpdates(keys)
+	if err != nil {
+		sendResponse(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	sendResponse(w, http.StatusOK, subscriptionId)
+}
+
+// stopSubscription is the handler for unsubscribing from config store
+func stopSubscription(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	subscriptionId := vars["subscriptionId"]
+	url := "http://" + daprHost + ":" + daprHTTPPort + "/v1.0-alpha1/configuration/" + configStoreName + "/" + subscriptionId + "/unsubscribe"
+	resp, err := http.Get(url)
+	if err != nil {
+		sendResponse(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	defer resp.Body.Close()
+	respInBytes, _ := ioutil.ReadAll(resp.Body)
+	sendResponse(w, http.StatusOK, string(respInBytes))
+}
+
+// configurationUpdateHandler is the handler for receiving updates from config store
+func configurationUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	var updateEvent configuration.UpdateEvent
+	err := json.NewDecoder(r.Body).Decode(&updateEvent)
+	if err != nil {
+		sendResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	subId := updateEvent.ID
+	receivedItemsInBytes, err := json.Marshal(updateEvent.Items)
+	receivedItems := string(receivedItemsInBytes)
+
+	log.Printf("SubscriptionID: %s, Received item: %s\n", subId, receivedItems)
+	lock.Lock()
+	defer lock.Unlock()
+	if receivedUpdates[subId] == nil {
+		receivedUpdates[subId] = make([]string, 10)
+	}
+	receivedUpdates[subId] = append(receivedUpdates[subId], receivedItems)
+	sendResponse(w, http.StatusOK, "OK")
+}
+
+// getReceivedUpdates is the handler for getting received updates from config store
+func getReceivedUpdates(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	subId := vars["subscriptionId"]
+
+	lock.Lock()
+	defer lock.Unlock()
+	response := receivedMessagesResponse{
+		ReceivedUpdates: receivedUpdates[subId],
+	}
+	receivedUpdates[subId] = make([]string, 10)
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(response)
+}
+
+// initializeUpdater is the handler for initializing config updater
+func initializeUpdater(w http.ResponseWriter, r *http.Request) {
+	var component string
+	err := json.NewDecoder(r.Body).Decode(&component)
+	if err != nil {
+		sendResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	log.Printf("Initializing updater with component: %s\n", component)
+	switch component {
+	case "redis":
+		updater = cu_redis.NewRedisConfigUpdater(logger.NewLogger("configuration.redis"))
+		err := updater.Init(map[string]string{
```

Comment:
I think E2E tests should not depend on Dapr code (incl. components-contrib) directly. The purpose of E2E tests is to validate the experience… E2E, which means going through the Dapr sidecar for everything.

Reply:
That is the ideal way agreed. But in configuration API, dapr sidecar just has read access to the config store. So there is no add/update/delete api exposed through dapr and that's why we need the updater interface for updating the key-value pairs.

Reply:
Please, let's find another way to do that, even if that involves connecting to Redis directly.

I would really, really prefer to avoid adding another circular dependency, as that will cause headaches down the road :)

Reply:
Done. Please check @ItalyPaleAle. Thanks